posts/2021-11-26-jakarta-ee-9.1.adoc

```diff
@@ -0,0 +1,433 @@
+	at com.ibm.ws.tcpchannel.internal.NBAcceptChannelSelector.performRequest(NBAcceptChannelSelector.java:358)
+	at com.ibm.ws.tcpchannel.internal.ChannelSelector.run(ChannelSelector.java:257)
+	at java.base/java.lang.Thread.run(Thread.java:866)
+Caused by: java.nio.channels.NotYetBoundException
+	at java.base/sun.nio.ch.ServerSocketChannelImpl.begin(ServerSocketChannelImpl.java:246)
+	at java.base/sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:283)
+	at com.ibm.ws.tcpchannel.internal.NBAcceptChannelSelector.performRequest(NBAcceptChannelSelector.java:278)
+	... 2 more
+----
++
+This issue has now been fixed so that all configured ports should start, or if there is a problem some meaningful error message should be logged.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18992[Application fails to restart in server.xml update scenario]
++
+We discovered an issue where an application would fail to restart, due to a race condition during server reconfiguration when multiple apps are installed. The problem occurs when one app starts before another app is finished uninstalling. In theory this shouldn't be a problem - however for this scenario these apps are sharing a VirtualHost configuration object, and in this case one app updates the parent VirtualHost as part of its uninstall process in such a way that the other gets into an invalid state. The server log will show an warning such as `CWWKZ0020I: Application <app_name> not updated`. This issue was fixed by fixing the race condition that caused the failure.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17599[HTTP upgrade to WebSocket can cause quiesce errors]
++
+When a websocket connection is started, it starts as an HTTP connection. Previously, If an error occured during the transition between an HTTP and a websocket connection, which was known to be a websocket upgrade, the error processing would neglect to decrement a connection counter, which then caused the server to believe there is an open connection during server stop.
+There were two scenarios where these quiesce errors would occur:
++
+** When a read error occured during the transition between an HTTP and a websocket connection, the error processing neglected to decrement a connection counter, which then causes the server to believe there is an open connection during server stop.
++
+** If a client immediately closed the websocket connection after it was opened, the original upgrade request handling may not have had enough time to close properly on the server. Once again, the connection counter failed to decrement leading the server to believe there is an open connection during the server stop.
++
+This issue has been fixed by adding a new flag called `decrementNeeded` which helps to ensure that the decrement is not neglected.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/19193[Stop allowing creation of HTTP/2 streams if server is closing]
```

Comment:
I think we should omit this issue completely - a partial fix was put in. I haven't created a new issue that will completely resolve this issue, but we do have an internal [PE/ZE issue](https://github.ibm.com/WASL3/MDV/issues/79) open.

Reply:
removed